### ISSUE TITLE
EFI filename extension: Uppercase or lowercase?

### ISSUE THREAD

**Issue by mk-pmb** [supergrub/supergrub#42]
I split this out from #40 because I think it can grow a rather large endaevor.

* efibootmgr thinks lowercase is ok:
  * ["The default OS Loader is elilo.efi."](https://github.com/rhboot/efibootmgr)
  * [Gentoo Wiki](https://wiki.gentoo.org/wiki/Efibootmgr) uses lowercase in all file name examples. However, they do use uppercase for the EFI directory itself.
* [Wikipedia: Extensible_Firmware_Interface](https://en.wikipedia.org/wiki/Extensible_Firmware_Interface) is undecided:
  * `BOOT<MACHINE_TYPE_SHORT_NAME>.EFI` suggests full path all-uppercase for at least the default images.
  * `/efi/BOOT/BOOTX64.EFI` (albeit I read somewhere else the EFI directory must be uppercase)
  * `efi\boot\bootaa64.efi`
* [rEFInd](http://www.rodsbooks.com/refind/configfile.html) claims: "EFI loader filename extensions are coded as lowercase .efi. I made these choices because they seem to be the most common uses on real-world installations."
* [UEFI Specs](https://uefi.org/specifications)
  * UEFI spec v2.0 (2006-jan), section 3.4.1.1 "Removable Media Boot Behavior",
    Table 9, says `BOOTIA32.EFI`, `BOOTx64.EFI` (lower case x, as opposed to WP!), `BOOTIA64.EFI`
    * [List of where case stuff is mentioned](https://gist.github.com/mk-pmb/d33d907a6d57ca1fc095a603589aa917) (I wrote a bunch of arcane sed magic to make the text usefully grep-able.)

**Comment by adrian15** [comment]
Well, you can check: https://sourceforge.net/projects/supergrub2/files/2.04s2-beta2/super_grub2_disk_2.04s2-beta2/supergrub2-2.04s2-beta2-multiarch-USB.img.zip/download to see what GNU/GRUB does by default.

Both capital letters EFI directory and its BOOTX64.EFI and BOOTIA32.EFI files.

**Comment by adrian15** [comment]
TL;DR: Both of them.
